### PR TITLE
further integration

### PR DESCRIPTION
**Pull request by pepyakin**
this change fleshes out the db opening and creation, hooks up the
writeout module and uncomments reconstruction (although without fixing
it properly)

### CODE REVIEW

**Comment by pepyakin** [comment]
* **#42** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/42?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#43](https://github.com/thrumdev/bitbox/pull/43) <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#44](https://github.com/thrumdev/bitbox/pull/44) <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/44?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#41** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/41?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pepyakin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/beatree/ops/reconstruction.rs [hunk 1]
```diff
@@ -1,136 +1,95 @@
...
+
+            let mut separator = [0u8; 32];
+            {
+                let prefix = view.prefix();
+                let separator = separator.view_bits_mut::<Lsb0>();
+                separator[..prefix.len()].copy_from_bitslice(prefix);
+                let first = view.separator(0);
+                separator[prefix.len()..prefix.len() + first.len()].copy_from_bitslice(first);
+            }
+            if let Some(_displaced_branch_id) = index.insert(separator, new_branch_id) {
```
Comment (rphmeier) [inline src/beatree/ops/reconstruction.rs:68]: the displacement logic is no longer necessary given that we have the freelist loaded up-front.

**Review by gab8i — APPROVED** [review]

File: src/beatree/ops/reconstruction.rs (hunk 1 above)
Comment (gab8i) [reply src/beatree/ops/reconstruction.rs:68]: should be updated in #44

**Comment by pepyakin** [comment]
### Merge activity

* **Jul 24, 6:10 AM EDT**: @pepyakin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/42).
* **Jul 24, 6:12 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/42) rebased this pull request as part of a merge.
* **Jul 24, 6:13 AM EDT**: @pepyakin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/42).

**Pull request merged**